### ISSUE TITLE
Use of `libtool` makes the Mac build noisy

### ISSUE THREAD

**Issue by judah** [tweag/rules_haskell#389]
When building a library on Mac like `mtl` that has several modules with the same "basename", libtool spams the output with unnecessary warnings.

For example, with Hazel, `bazel build @haskell_mtl//:mtl` produces:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Class.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Writer/Class.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Cont/Class.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Class.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Cont/Class.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/State/Class.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Class.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/State/Class.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Reader/Class.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Class.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Reader/Class.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/RWS/Class.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Class.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/RWS/Class.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Error/Class.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Lazy.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/State/Lazy.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Writer/Lazy.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Lazy.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Writer/Lazy.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/RWS/Lazy.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Strict.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/RWS/Strict.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/State/Strict.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (Strict.dyn_o) in output file used for input files: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/State/Strict.dyn_o and: bazel-out/darwin-fastbuild/bin/external/haskell_mtl/_obj/mtl/Control/Monad/Writer/Strict.dyn_o (due to use of basename, truncation, blank padding or duplicate input files)
```

This started happening when we switched from `ar` to `libtool` in #383.  (It looks like `ar` still accepts them.)  Given that even Cabal still uses `ar`, I think it would be reasonable at this time for us to also use `ar`.  Long-term maybe we can mangle the files further (perhaps wait to see if/how Cabal resolves the issue).

Unfortunately, it looks like the Mac `ar` doesn't support filelist arguments so I think we'd need to use a `run_shell` to pipe the filelist into the arguments of `ar`.

**Comment by mboes** [comment]
Confusingly, despite the observations in https://github.com/bazelbuild/bazel/issues/5127, what I'm observing on CircleCI's OSX instances is that `wrapped_ar` (which in turn calls `ar`) is being used for assembling archives. I don't care much what Cabal uses, but if even the CC rules use `ar`, despite the purported advantages (faster, not deprecated), and libtool is causing problems, then let's just use `ar`.

Incidentally, I would petition for a reopen of https://github.com/bazelbuild/bazel/issues/5127. Because the current logic seems bizarre. Either `ar` is just fine and it should be selected always by the CC rules, or it's not in which case it should always be libtool.
